core/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalNumericMetricsAggregation.java

```diff
@@ -122,4 +135,48 @@ public boolean equals(Object obj) {
         return super.equals(obj) &&
                 Objects.equals(format, other.format);
     }
+
+    protected static DocValueFormat wrapIntoDocValueFormat(Map<Object , String> valuesAsStrings) {
+        return new DocValueFormat() {
+
+            @Override
+            public String getWriteableName() {
+                return null;
+            }
+
+            @Override
+            public void writeTo(StreamOutput out) throws IOException {
+            }
+
+            @Override
+            public String format(long value) {
+                return valuesAsStrings.getOrDefault(value, DEFAULT_FORMAT.format(value));
+            }
+
+            @Override
+            public String format(double value) {
+                return valuesAsStrings.getOrDefault(value, DEFAULT_FORMAT.format(value));
+            }
+
+            @Override
+            public String format(BytesRef value) {
+                return valuesAsStrings.getOrDefault(value, DEFAULT_FORMAT.format(value));
+            }
+
+            @Override
+            public long parseLong(String value, boolean roundUp, LongSupplier now) {
+                return 0;
+            }
+
+            @Override
+            public double parseDouble(String value, boolean roundUp, LongSupplier now) {
+                return 0;
+            }
+
+            @Override
+            public BytesRef parseBytesRef(String value) {
+                return null;
+            }
```

Comment:
I like the impl from the other PR better as it was throwing UOE in all these methods.